Source/EmbeddedBoundary/WarpXInitEB.cpp

```diff
@@ -15,8 +15,162 @@ WarpX::InitEB ()
 
     amrex::ParmParse pp_eb2("eb2");
     if (!pp_eb2.contains("geom_type")) {
-        pp_eb2.add("geom_type", "all_regular"); // use all_regular by default
+        std::string geom_type = "all_regular";
+        pp_eb2.add("geom_type", geom_type); // use all_regular by default
     }
     amrex::EB2::Build(Geom(maxLevel()), maxLevel(), maxLevel());
+
+#endif
+}
+
+/**
+ * \brief Compute the length of the mesh edges. Here the length is a value in [0, 1].
+ *        An edge of length 0 is fully covered.
+ */
+void
+WarpX::ComputeEdgeLengths () {
+#ifdef AMREX_USE_EB
+    BL_PROFILE("ComputeEdgeLengths");
+
+    auto const eb_fact = fieldEBFactory(maxLevel());
+
+    auto const &flags = eb_fact.getMultiEBCellFlagFab();
+    auto const &edge_centroid = eb_fact.getEdgeCent();
+    for (amrex::MFIter mfi(flags); mfi.isValid(); ++mfi){
+        amrex::Box const &box = mfi.validbox();
+        amrex::FabType fab_type = flags[mfi].getType(box);
+        for (int idim = 0; idim < AMREX_SPACEDIM; ++idim){
+            auto const &edge_lengths_dim = m_edge_lengths[maxLevel()][idim]->array(mfi);
+            if (fab_type == amrex::FabType::regular) {
+                // every cell in box is all regular
+                amrex::LoopOnCpu(amrex::convert(box, amrex::Box(edge_lengths_dim).ixType()),
+                                 [=](int i, int j, int k) {
+                    edge_lengths_dim(i, j, k) = 1.;
+                });
+            } else if (fab_type == amrex::FabType::covered) {
+                // every cell in box is all covered
+                amrex::LoopOnCpu(amrex::convert(box, amrex::Box(edge_lengths_dim).ixType()),
+                                 [=](int i, int j, int k) {
+                    edge_lengths_dim(i, j, k) = 1.;
```

Comment:
Should this be
```suggestion
                    edge_lengths_dim(i, j, k) = 0.;
```

Reply:
That was indeed a mistake. Thanks a lot.